uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/BuildForm/OptionEditPathFilter.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+class CRM_HRCore_Hook_BuildForm_OptionEditPathFilter {
+
+  /**
+   * @var bool[]
+   *   A mapping of option group name to its "is_locked" status
+   */
+  private $optionGroupLockedStatuses = [];
+
+  /**
+   * Handle the form. Any form that can contain elements referencing an
+   * option group should be handled, i.e. all forms
+   *
+   * @param string $formName
+   * @param CRM_Core_Form $form
+   */
+  public function handle($formName, &$form) {
+    $this->filterOptionEditPaths($form);
+  }
+
+  /**
+   * Loop through the form, removing the 'data-option-edit-path' from form
+   * elements that reference locked option groups. If this attribute is not set
+   * then the icon to provide a shortcut to edit this group will not be shown.
+   *
+   * @see \CRM_Core_Form_Renderer::addOptionsEditLink
+   *
+   * @param CRM_Core_Form $form
+   */
+  private function filterOptionEditPaths($form) {
+    /** @var HTML_QuickForm_element $element */
+    foreach ($form->_elements as &$element) {
+      $optionEditPath = $element->getAttribute('data-option-edit-path');
+      if (!$optionEditPath) {
+        continue;
+      }
+
+      $prefix = 'civicrm/admin/options/';
+      if (strpos($optionEditPath, $prefix) !== 0) {
+        // this is not an option group edit link
+        continue;
+      }
```

Comment:
Is this path check really necessary? As we can see [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Form.php?utf8=%E2%9C%93#L1332) and [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Form.php?utf8=%E2%9C%93#L1349), it's possible to customize this URL using the `option_url` property. Also, if you check [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/PseudoConstant.php#L483-L495) it looks like there's an edge case scenario for when the optionGroup name is not available in the field metadata. In all of these cases, the might not start with `civicrm/admin/options/`. Should we care about that?

Reply:
Based on your investigation, it looks like that `option_url` is used when there a need to point to something that is not really an option group. The edge case also seems to be used for similar purposes. In this case, your solution is good enough as is. No need for changes.